### PR TITLE
feat: `/me` endpoint for v2 api

### PR DESCRIPTION
**Pull request by Ryukemeister**
## What does this PR do?

Adds a `/me` endpoint in the v2 api that returns some general info about a user

### CODE REVIEW

File: apps/api/v2/src/modules/user/controllers/user.controller.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+export class UserController {
+  private readonly logger = new Logger("UserController");
+
+  constructor(private readonly userRepository: UserRepository) {}
+
+  @Get("/")
+  @HttpCode(HttpStatus.OK)
+  @UseGuards(AccessTokenGuard)
+  async getUser(
+    @Param("clientId") clientId: string,
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/user/controllers/user.controller.ts:24]: client_id is not needed

File: apps/api/v2/src/modules/user/controllers/user.controller.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    @Body() body: GetUserInput,
+    @GetUser() user: any
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/user/controllers/user.controller.ts:26]: avoid using any, use the type return by getUser decorator

File: apps/api/v2/src/modules/user/controllers/user.controller.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  ): Promise<ApiResponse<UserReturned>> {
+    if (!user) {
+      throw new BadRequestException("This user does not exist.");
+    }
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/user/controllers/user.controller.ts:30]: not needed, the access token guard should fail if there are no associated user

**Comment by Ryukemeister** [comment]
Closing this since everything here is included in #13006

**Pull request closed without merging**